### ISSUE TITLE
Update flutter/packages macOS stable tests to Mac-13 or Mac-14

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#149830]
The following tests can't be run on macOS 14 until https://github.com/flutter/flutter/pull/149618 lands in stable and the stable version in flutter/packages has been updated to a version that contains the fix:

* Mac_arm64 macos_platform_tests stable - packages
* Mac_arm64 custom_package_tests stable

**Comment by jmagman** [comment]
I think these are builds that show those macOS sandbox-related failures:
https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20custom_package_tests%20stable/2/overview
https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20macos_platform_tests%20master%20-%20packages/2/overview

@vashworth instead, can we disable the sandbox for the `url_launcher_macos` and `pigeon` example app, then have a TODO and tracking issue to turn it back on when that tools change gets to stable?

**Comment by vashworth** [comment]
> I think these are builds that show those macOS sandbox-related failures: https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20custom_package_tests%20stable/2/overview https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20macos_platform_tests%20master%20-%20packages/2/overview
> 
> @vashworth instead, can we disable the sandbox for the `url_launcher_macos` and `pigeon` example app, then have a TODO and tracking issue to turn it back on when that tools change gets to stable?

Sure, I'll try that

**Comment by vashworth** [comment]
> I think these are builds that show those macOS sandbox-related failures: https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20custom_package_tests%20stable/2/overview https://ci.chromium.org/ui/p/flutter/builders/staging.shadow/Mac_arm64%20macos_platform_tests%20master%20-%20packages/2/overview
> 
> @vashworth instead, can we disable the sandbox for the `url_launcher_macos` and `pigeon` example app, then have a TODO and tracking issue to turn it back on when that tools change gets to stable?

So doing this for pigeon will fix the `custom_package_tests stable`.

However to fix for `macos_platform_tests stable`, we'd actually have to disable sandboxing for the following package example apps: flutter_image, google_sign_in, google_sign_in_ios, image_picker, in_app_purchase, in_app_purchase_storekit, path_provider, path_provider_foundation, shared_preferences, shared_preferences_foundation, url_launcher, url_launcher_macos, video_player, video_player_avfoundation. The test failure doesn't show because it fails before it reaches, but there's a [later part of the test that does `flutter test -d macos` on each of those](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8745866474214880257/+/u/Run_package_tests/drive_examples/stdout), which will fail without https://github.com/flutter/flutter/pull/149618. 

@jmagman Do you think it'd still be better to disable them directly rather than just keep them on Mac-13 for now? I think it's probably fine and chances of a new one being added are super low, so I'm okay to just directly disable all of them for now, but it is a lot more than just url_launcher_macos.

**Comment by jmagman** [comment]
I think we should disable sandboxing on all of them temporarily, as long as we have an action item to turn it back on after next stable.
@stuartmorgan do you have any concerns about this?

**Comment by jmagman** [comment]
It's definitely not ideal, but it would require more work to clear out the app containers between test runs (some more details in the referenced PR https://github.com/flutter/flutter/pull/149618).  I think turning the sandbox off for tests is the best bad solution we have to unblock the macOS 14 update... 😞

**Comment by vashworth** [comment]
Closing in favor of https://github.com/flutter/flutter/issues/149844
